### PR TITLE
fix(frontend): show clear preview unavailable message for office documents

### LINKED ISSUE #953: [Bug]: Preview tab shows blank screen for pptx files
### Operating System

macOS

### Installation Method

npm (`npx @openhands/agent-canvas` or `npm install -g @openhands/agent-canvas`)

### Agent Canvas Version

main

### Bug Description

When opening a pptx file in the preview tab, nothing shows up but also no error is shown.

### Steps to Reproduce

Open a pptx file in the preview tab.

### Actual Behavior

Nothing is shown.

### Expected Behavior

Either the pptx shows up or an error says that this file type cannot be shown.

### Relevant Logs

```shell

```

### Screenshots

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by hieptl**
- [x] A human has tested these changes.

---

## Problem

When a user opens a `.pptx` file in a conversation's **Files** tab, the preview pane shows only the generic **"Binary file – preview not available"** message (and in some flows reads as an empty pane). Nothing indicates that the file is a PowerPoint document or why it can't be shown.

## Steps to reproduce

1. Start the dev server (`npm run dev`).
2. Create a conversation from a workspace that already contains `demo.pptx`.
3. Open the conversation → **Files** tab → select `demo.pptx`.

## Current behavior

The preview pane shows the generic "Binary file – preview not available" string for `.pptx` (and other Office documents) — no useful information.

## Expected behavior

Either render the document **or** show a clear message that the format isn't supported for preview, e.g. _"Preview isn't available for PowerPoint files."_

## Root cause

The preview pipeline only recognizes images and PDFs. `.pptx` falls through `classifyKind()` to `"text"`; after the bytes are fetched, `isLikelyBinary()` finds NUL bytes in the ZIP header and flips it to `kind: "binary"`; and `FileContentViewer` renders a single generic message for every binary kind. There is no Office-document recognition or format-specific messaging.

## Proposed solution

Recognize common Office/document extensions (`.pptx, .ppt, .docx, .doc, .xlsx, .xls`) in `FileContentViewer` and render a clear, format-named message via a new `FILES$UNSUPPORTED_DOCUMENT` i18n key. Keep the existing generic message for all other binaries. Rendering the actual slide/document contents (which would need a heavy client-side dependency) is out of scope.

## Acceptance criteria

- [ ] Previewing `.pptx` / `.docx` / `.xlsx` shows a clear "Preview isn't available for {PowerPoint|Word|Excel} files." message in both Rich and Plain modes.
- [ ] The preview pane is never blank for these files.
- [ ] Other binary files still show the existing generic message.
- [ ] A unit test covers the Office-document case.

## Summary

Office documents (`.pptx`, `.docx`, `.xlsx`, …) opened in the **Files** tab previously showed only the generic _"Binary file – preview not available"_ message (and could read as a blank pane). They now show a clear, format-aware message — e.g. **"Preview isn't available for PowerPoint files."** — so the limitation is explicit and the pane is never blank.

## Root cause

The preview pipeline only classifies images and PDFs specially:

- `classifyKind()` (`use-workspace-file-content.ts`) sends `.pptx` to `"text"`.
- After fetching, `isLikelyBinary()` finds NUL bytes in the ZIP header → `kind: "binary"`.
- `FileContentViewer` renders a single generic message for every `"binary"` file.

So Office documents are treated as anonymous binaries with no format-specific messaging.

## What changed

- **`file-content-viewer.tsx`** — added an `OFFICE_DOCUMENT_LABELS` map (which doubles as the allow-list) and a small `UnpreviewableFallback` helper that shows a clear, format-named message for Office docs and the existing generic message for any other binary. The three previously-duplicated fallback blocks (plain mode, rich-mode binary branch, and final fallback) now route through this helper, so the message is consistent across Rich/Plain and the pane is never blank.
- **`i18n/translation.json`** — new `FILES$UNSUPPORTED_DOCUMENT` key (all 15 locales, `{{type}}` placeholder). `declaration.ts` is regenerated from this by `make-i18n` (gitignored).
- **Test** — new `__tests__/components/features/files-tab/file-content-viewer.test.tsx` drives the real `useWorkspaceFileContent` hook with the underlying `fetch` service mocked to return `.pptx` bytes, and asserts the clear message renders.

The data layer (`classifyKind`, `WorkspaceFileKind`) is unchanged — real Office docs already resolve to `"binary"`.

## Behavior

| File                          | Before                                | After                                           |
| ----------------------------- | ------------------------------------- | ----------------------------------------------- |
| `demo.pptx`                   | "Binary file – preview not available" | "Preview isn't available for PowerPoint files." |
| `notes.docx`                  | "Binary file – preview not available" | "Preview isn't available for Word files."       |
| `data.xlsx`                   | "Binary file – preview not available" | "Preview isn't available for Excel files."      |
| other binary (e.g. `app.bin`) | "Binary file – preview not available" | unchanged                                       |

## Testing

- `npm run typecheck` — clean
- New unit test passes; the 25 existing tests in `files-tab.test.tsx` and `use-workspace-file-content.test.tsx` remain green
- ESLint + Prettier clean on changed files

## Out of scope

- Rendering the actual slide/document contents (would require a heavy client-side OOXML renderer).
- A download / "open raw" affordance (the `staticUrl` is available if we want to add one later).

## Issue Number

Resolves #953 

## How to Test

1. Clone the `agent-server-gui` repository and start the development server:

```bash
npm run dev
```

2. From the home page, create a new conversation using a workspace that already contains a file named `demo.pptx`.

3. Open the conversation, navigate to the **Files** tab, and select `demo.pptx` to preview it.

## Video/Screenshots

app-2069

## Type

- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Breaking change
- [ ] Docs / chore

---
**🐳 Docker images for this PR**

• **GHCR package:** https://github.com/OpenHands/agent-canvas/pkgs/container/agent-canvas

| Component | Value |
|---|---|
| **Image** | `ghcr.io/openhands/agent-canvas` |
| **Architectures** | amd64, arm64 |
| **Agent Server** | `ghcr.io/openhands/agent-server:1.24.0-python` |
| **Automation** | `openhands-automation==1.0.0a5` |
| **Commit** | `d91e9a5559cbf845c63ed3fe7f9c69aef0d6c413` |

**Pull (multi-arch manifest)**
```bash
# Multi-arch manifest — Docker automatically pulls the correct architecture
docker pull ghcr.io/openhands/agent-canvas:sha-d91e9a5
```

**Run**
```bash
docker run -it --rm \
  -p 8000:8000 \
  ghcr.io/openhands/agent-canvas:sha-d91e9a5
```

**All tags pushed for this build**
```
ghcr.io/openhands/agent-canvas:sha-d91e9a5-amd64
ghcr.io/openhands/agent-canvas:hieptl-app-2069-amd64
ghcr.io/openhands/agent-canvas:pr-958-amd64
ghcr.io/openhands/agent-canvas:sha-d91e9a5-arm64
ghcr.io/openhands/agent-canvas:hieptl-app-2069-arm64
ghcr.io/openhands/agent-canvas:pr-958-arm64
ghcr.io/openhands/agent-canvas:sha-d91e9a5
ghcr.io/openhands/agent-canvas:hieptl-app-2069
ghcr.io/openhands/agent-canvas:pr-958
```

**About Multi-Architecture Support**
- Each tag (e.g., `sha-d91e9a5`) is a **multi-arch manifest** supporting both **amd64** and **arm64**
- Docker automatically pulls the correct architecture for your platform
- Individual architecture tags (e.g., `sha-d91e9a5-amd64`) are also available if needed

### CODE REVIEW

File: __tests__/components/features/files-tab/file-content-viewer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      orgId: null,
+    });
+    useWorkspaceMutationCounter.setState({ count: 0 });
+  });
+
+  afterEach(() => {
+    vi.unstubAllGlobals();
+  });
+
+  it("shows a clear unsupported-document message when previewing an Office file (.pptx)", async () => {
```
[the remark this answers is not in the document]
Comment (hieptl) [reply __tests__/components/features/files-tab/file-content-viewer.test.tsx]: ### Analysis
The concern is legitimate (guard the new conditional `data-testid`), but the distinction is **already guarded from both directions**, so a dedicated test here would duplicate existing coverage:

- **Non-Office → generic direction:** `__tests__/routes/files-tab.test.tsx:219` ("renders the binary fallback in plain mode for binary files") renders the **real** `FileContentViewer`/`UnpreviewableFallback` (that file mocks only the *hook*, not the component) with `path: "logo.png"`, and asserts `file-content-viewer-binary-fallback`. If `UnpreviewableFallback` ever emitted the Office testid for a non-Office path, this test fails.
- **Office → unsupported direction:** the new test (Item 1) asserts `file-content-viewer-unsupported-document` for `.pptx`. If Office detection were removed, this fails.
- **Real pipeline classifies a `.bin` as binary:** `__tests__/hooks/query/use-workspace-file-content.test.tsx:166` ("flips text → binary when the fetched bytes contain a NUL", `data/blob.bin`) already covers the classification half.

The proposed `app.bin` test is exactly the union of those already-covered halves. Adding it conflicts with the repo's testing guidance — *"Avoid duplicating test cases or logic," "Do not assert the same condition more than once,"* and *"keep the number of test cases to the minimum necessary."* The reviewer also flagged this as optional ("can be addressed in a follow-up if the acceptance criteria is a hard requirement").

> Note: an `app.bin` testid assertion would *not* catch a wrong-message regression either (e.g. a swapped i18n key), since it asserts the testid, not the text — so it adds no protection beyond what exists.

### Decision
No change. If the team later prefers the office-vs-generic pair co-located in this unit file for self-documentation, it's a trivial add — but it is not additional coverage today.

File: __tests__/components/features/files-tab/file-content-viewer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+}));
+
+const useRuntimeIsReadyMock = vi.fn();
+vi.mock("#/hooks/use-runtime-is-ready", () => ({
+  useRuntimeIsReady: () => useRuntimeIsReadyMock(),
+}));
+
+const getActiveBackendMock = vi.fn();
+vi.mock("#/api/backend-registry/active-store", () => ({
+  getActiveBackend: () => getActiveBackendMock(),
```
[the remark this answers is not in the document]
Comment (hieptl) [reply __tests__/components/features/files-tab/file-content-viewer.test.tsx:37]: ### Analysis
Correct. The test configures a **local** backend (`getActiveBackend → kind: "local"`), so `useWorkspaceFileContent` takes the `fetch` branch; `downloadFile` (the **cloud** branch) is never called or asserted. The named `downloadFileMock` handle and its `mockReset()` were dead weight.

One nuance worth calling out: I did **not** delete the `vi.mock(...)` for the module. The hook *statically* imports `AgentServerRuntimeService` (`use-workspace-file-content.ts:4`), and that module pulls in `@openhands/typescript-client` clients, `RemoteWorkspace`, and `callCloudProxy` at import time. Unmocking it would load real cloud/proxy machinery into a local-path unit test — which is exactly why the sibling hook test mocks it too. So the right fix is to keep a **minimal module stub** and drop only the unused handle.

### What changed
Removed the `downloadFileMock` const and its `mockReset()` line; inlined an anonymous `vi.fn()` in the mock factory and added a one-line note explaining why the module stays stubbed.

```diff
-const downloadFileMock = vi.fn();
-vi.mock("#/api/runtime-service/agent-server-runtime-service", () => ({
-  default: {
-    downloadFile: (...args: unknown[]) => downloadFileMock(...args),
-  },
-}));
+// The hook statically imports the cloud runtime service; stub the module so
+// this local-path test never loads the real cloud/proxy machinery. The test
+// uses the fetch (local) path, so downloadFile is never called or asserted.
+vi.mock("#/api/runtime-service/agent-server-runtime-service", () => ({
+  default: { downloadFile: vi.fn() },
+}));
```
```diff
     getActiveBackendMock.mockReset();
-    downloadFileMock.mockReset();
```

Both parametrized cases (Rich + Plain) still pass.

**Comment by hieptl** [comment]
@all-hands-bot If the PR looks good, please go ahead and approve it.

**Comment by hieptl** [comment]
>Testing gap — non-Office binary regression: The new file-content-viewer.test.tsx asserts that .pptx shows data-testid="file-content-viewer-unsupported-document" but has no test asserting that a non-Office binary (e.g., app.bin) still renders data-testid="file-content-viewer-binary-fallback". The data-testid attribute changed from a static value to a conditional expression — a conditional that has two outputs. Only one output is exercised in the new test file. If the existing suite in files-tab.test.tsx already covers this case, the gap is covered; otherwise a one-liner using it.each on a non-Office path would make coverage airtight.

*Already covered: `files-tab.test.tsx`'s "binary fallback in plain mode" test renders the real `UnpreviewableFallback` for a non-Office binary (`logo.png`) and asserts `file-content-viewer-binary-fallback` (line 245), so the conditional's generic branch is guarded; the `.pptx` spec guards the Office branch.*

---

> Only .pptx covered — acceptable by design: The parametrised spec runs .pptx only. Since all Office extensions share the same code path through OFFICE_DOCUMENT_LABELS, the omission is low risk, but a quick it.each over ["demo.docx", "data.xlsx"] would directly satisfy the acceptance criteria's wording (".pptx / .docx / .xlsx"). Worth a follow-up if the test suite is ever expanded.

*Left as-is: all Office extensions go through the same data-driven `OFFICE_DOCUMENT_LABELS` lookup that `.pptx` already exercises, and because the test suite mocks i18n to return the key (ignoring the `{{type}}` label), a `.docx`/`.xlsx` case asserts the byte-identical outcome — same testid, same text, same branch — which trips our "no duplicate assertions" rule. If we want per-format label verification, the clean follow-up is a tiny unit test over the label map (it'd need to be exported), not a duplicated component spec.*

**Review by tofarr — APPROVED** [review]
🍰
